packages/graphql/src/schema/validation/enrichers/directive/utils.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Kind } from "graphql";
+import type {
+    ObjectTypeDefinitionNode,
+    FieldDefinitionNode,
+    InterfaceTypeDefinitionNode,
+    ObjectTypeExtensionNode,
+    InterfaceTypeExtensionNode,
+    GraphQLSchema,
+} from "graphql";
+
+export type ObjectOrInterfaceDefinitionNode = ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode;
+export type ObjectOrInterfaceExtensionNode = ObjectTypeExtensionNode | InterfaceTypeExtensionNode;
+type ObjectLikeDefinitionNode = ObjectOrInterfaceDefinitionNode | ObjectOrInterfaceExtensionNode;
+
+export type DIRECTIVE_TRANSFORM_FN = (currentDirectiveDirective: any, typeName: string) => any;
+export type CREATE_DIRECTIVE_DEFINITION_FN = (typeDefinitionName: string, schema: GraphQLSchema) => any;
```

Comment:
What are these types? Should they live here?

Reply:
They are passed to the enricher functions, for example the DIRECTIVE_TRANSFORM_FN  is passed as an argument to `directiveEnricher` which will call it with the encountered directive to transform it into the directive used as enrichment.
I thought it was better to have them explicitly defined although I see I forgot to use them in the enrichers/directive files..

Reply:
usage updated